### DISCUSSION TITLE
Load MusicXML from nameless JavaScript Blob-URL

### DISCUSSION THREAD

**Discussion by mirkosertic** [CoderLine/alphaTab#555] [Q&A] [answered]
Hi@all,
i am doing some MusixXML tweaking using XPath on document level. When I'm done, I store the generated MusicXML file as a Blob, and retrieve the URL by calling `window.URL.createObjectURL(blob)`. Now I pass this URL to alphaTab to load the file. Now the generated URL only contains some kind of content hash without a filename extension, and I am getting some strange errors while loading. Maybe alphaTab cannot resolve the correct importer, as there is no filename extension available? How can I tell alphaTab so interpret the URL as MusicXML? Is there some example code available? I tried to figure out by reading the documentation, but I am pretty sure I missed something, but I don't know where.
Thanks in advance,
Mirko

**Comment by Danielku15** [comment, marked as answer]
Hi there.
Is there a particular reason why you create a blob? In theory alphaTab can load directly from [`ArrayBuffer` or `Uint8Array`](https://alphatab.net/docs/reference/api/load/#parameters). The `string` loading variant is more meant for remote resources that need to be loaded via AJAX. I haven't tried what happens when you trigger an XmlHttpRequest to a blob url but I guess it will fail. 

So instead of making a blob, you could serialize your XML document to a string, and then to an `Uint8Array`.  Maybe something like the following could work out for you (untested): 
```js
function loadMusicXml(xmlDocument) {
	const xmlString = new XMLSerializer().serializeToString(xmlDocument.documentElement);
	const raw = new TextEncoder().encode(xmlString); // gives you an Uint8Array with the UTF8 encoded text inside
	api.load(raw, [0]);
}
```

In general alphaTab does not care about file names, it takes the raw file data and attempts to parse it with the available importers. Every importer has some basic mechanism to detect whether it should attempt reading the rest of the data fully. So basically it's a trial and error  where the first importer that works wins.
